src/mantaray-v1.ts

```diff
@@ -0,0 +1,738 @@
+    if (!entry) {
+      this._hasEntry = false
+
+      return
+    }
+    checkReference(entry)
+
+    this._entry = entry
+    this._hasEntry = true
+
+    if (entry.length === 64) this._encEntry = true
+    this.makeDirty()
+  }
+
+  public get entry(): Reference | undefined {
+    return this._entry
+  }
+
+  public set obfuscationKey(obfuscationKey: Bytes<32>) {
+    if (!(obfuscationKey instanceof Uint8Array)) {
+      throw new Error('Given obfuscationKey is not an Uint8Array instance.')
+    }
+
+    if (obfuscationKey.length !== 32) {
+      throw new Error(`Wrong obfuscationKey length. Entry only can be 32 length in bytes`)
+    }
+
+    this._obfuscationKey = obfuscationKey
+    this.makeDirty()
+  }
+
+  public get obfuscationKey(): Bytes<32> {
+    return this._obfuscationKey
+  }
+
+  public set nodeMetadata(metadata: MetadataMapping | undefined) {
+    this._nodeMetadata = metadata
+    this.makeDirty()
+  }
+
+  public get nodeMetadata(): MetadataMapping | undefined {
+    return this._nodeMetadata
+  }
+
+  public set forkMetadata(metadata: MetadataMapping | undefined) {
+    this._forkMetadata = metadata
+    this.makeDirty()
+  }
+
+  public get forkMetadata(): MetadataMapping | undefined {
+    return this._forkMetadata
+  }
+
+  public set forkMetadataSegmentSize(value: number) {
+    if (value > 31) throw new Error(`forkMetadataSegmentSize is greater than 31. Got: ${value}`)
+    this._forkMetadataSegmentSize = value
+  }
+
+  public get forkMetadataSegmentSize(): number {
+    return this._forkMetadataSegmentSize
+  }
+
+  public get isEdge(): boolean {
+    return this._isEdge
+  }
+
+  public get hasEntry(): boolean {
+    return this._hasEntry
+  }
+
+  public get encEntry(): boolean {
+    return this._encEntry
+  }
+
+  public get metadata(): MetadataMapping | undefined {
+    if (!this._forkMetadata && !this._nodeMetadata) return undefined
+
+    return {
+      ...this.nodeMetadata,
+      ...this._forkMetadata,
+    }
+  }
+
+  /// Node type related functions
+  /// dirty flag is not necessary to be set
+
+  public isValueType(): boolean {
+    return this._hasEntry
+  }
+
+  /**
+   * The node either has metadata on node level or fork level
+   * for forkMetadata parent node has to be fetched
+   */
+  public isWithMetadataType(): boolean {
+    return Boolean(this._forkMetadata) || Boolean(this._nodeMetadata)
+  }
+
+  /// BL methods
+
+  public addFork(
+    path: Uint8Array,
+    attributes?: {
+      entry?: Reference
+      nodeMetadata?: MetadataMapping
+      forkMetadata?: MetadataMapping
+      obfuscationKeyGenerator?: Random32BytesFn
+    },
+  ): void {
+    const entry: Reference | undefined = attributes?.entry
+    const nodeMetadata: MetadataMapping | undefined = attributes?.nodeMetadata
+    const forkMetadata: MetadataMapping | undefined = attributes?.forkMetadata
+    const obfuscationKeyGenerator: Random32BytesFn | undefined = attributes?.obfuscationKeyGenerator
+
+    if (path.length === 0) {
```

Comment:
What does it mean when a path is empty? Could you please add explaining comment?

Reply:
commented in the code